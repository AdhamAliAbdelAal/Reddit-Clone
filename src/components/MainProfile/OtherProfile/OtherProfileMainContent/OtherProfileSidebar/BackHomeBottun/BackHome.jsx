import { Box } from '@mui/system';
// eslint-disable-next-line import/no-named-as-default
import BackHome from './style';

function BackHomeBottun() {
  function goToTop() {
    document.documentElement.scrollTop = 0;
    document.body.scrollTop = 0;
  }
  return (
    <Box sx={{
      position: 'relative', flex: '1 1 auto',
    }}
    >
      <BackHome
        variant="contained"
        fontSize={15}
        fontWeight="bold"
        onClick={() => goToTop()}
      >
        Back to Top
      </BackHome>
    </Box>
  );
}

export default BackHomeBottun;
