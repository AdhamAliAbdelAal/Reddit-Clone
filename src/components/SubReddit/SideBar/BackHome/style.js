import { Button } from '@mui/material';
import { styled } from '@mui/material/styles';

// eslint-disable-next-line import/prefer-default-export
export const BackHome = styled(Button)(({
  fontSize, padding, fontWeight, backgroundColor,
}) => ({
  boxShadow: 'none',
  backgroundColor,
  fontSize,
  padding,
  borderRadius: 20,
  textTransform: 'initial',
  minWidth: 'auto',
  fontWeight,
  top: 'calc(100vh - 8px)',
  transform: 'translateY(-100%)',
  justifyContent: 'center',
  textAlign: 'center',
  position: 'sticky',
  margin: 90,
}));
export default BackHome;
