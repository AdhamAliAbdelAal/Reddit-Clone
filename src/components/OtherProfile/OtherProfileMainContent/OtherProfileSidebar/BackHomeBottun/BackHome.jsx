// eslint-disable-next-line import/no-named-as-default
import BackHome from './style';

function BackHomeBottun() {
  function goToTop() {
    document.documentElement.scrollTop = 0;
    document.body.scrollTop = 0;
  }
  return (
    <BackHome
      variant="contained"
      fontSize={15}
      fontWeight="bold"
      onClick={() => goToTop()}
    >
      Back to Top
    </BackHome>
  );
}

export default BackHomeBottun;
