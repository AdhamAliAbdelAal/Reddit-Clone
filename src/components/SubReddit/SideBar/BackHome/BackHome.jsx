// eslint-disable-next-line import/no-named-as-default
import BackHome from './style';
/**
 * function to handle in click back to to of page
 * @returns top of the page
 */
function BackHomeBottun() {
  function goToTop() {
    document.documentElement.scrollTop = 0;
    document.body.scrollTop = 0;
  }
  /**
 * button on click on it back to top of bage
 * @return {React.Component} - button
 */
  return (
    <BackHome
      variant="contained"
      fontSize={15}
      fontWeight="bold"
      onClick={() => goToTop()}
    >
      Back to Top
    </BackHome>
  );
}

export default BackHomeBottun;
