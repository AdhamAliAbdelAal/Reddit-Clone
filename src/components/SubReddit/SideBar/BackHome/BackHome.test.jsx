import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import BackHome from './BackHome';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<BackHome />).toJSON();
  expect(tree).toMatchSnapshot();
});
