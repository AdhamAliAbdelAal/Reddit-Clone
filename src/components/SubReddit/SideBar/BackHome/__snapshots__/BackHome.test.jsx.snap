// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-ub553u-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  fontSize={15}
  fontWeight="bold"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Back to Top
</button>
`;
